javascriptv3/example_code/bedrock-runtime/scenarios/converse_tool_scenario/converse-tool-scenario.js

```diff
@@ -0,0 +1,314 @@
+async function ProcessModelResponseAsync(response, messages, max_recursions) {
+  if (max_recursions <= 0) {
+    await HandleToolUseAsync(response, messages);
+  }
+  if (response.stopReason === "tool_use") {
+    await HandleToolUseAsync(response, messages, max_recursions - 1);
+  }
+  if (response.stopReason === "end_turn") {
+    const messageToPrint = response.output.message.content[0].text;
+    console.log(messageToPrint.replace(/<[^>]+>/g, ""));
+  }
+}
+// Handles the tool use case by invoking the specified tool and sending the tool's response back to Bedrock.
+// The tool response is appended to the conversation, and the conversation is sent back to Amazon Bedrock for further processing.
+// param "response" - the model's response containing the tool use request.
+// param "messages" - the conversation history.
+// param "max_recursions" - The maximum number of recursive calls allowed.
+async function HandleToolUseAsync(response, messages, max_recursions) {
+  const toolResultFinal = [];
+  try {
+    const output_message = response.output.message;
+    messages.push(output_message);
+    const toolRequests = output_message.content;
+    const toolMessage = toolRequests[0].text;
+    console.log(toolMessage.replace(/<[^>]+>/g, ""));
+    for (const toolRequest of toolRequests) {
+      if (Object.hasOwn(toolRequest, "toolUse")) {
+        const toolUse = output_message.content[1].toolUse;
+        const toolUseID = toolUse.toolUseId;
+        const latitude = toolUse.input.latitude;
+        const longitude = toolUse.input.longitude;
+        console.log(
+          `Requesting tool ${toolUse.name}, Tool use id ${toolUse.toolUseId}`,
+        );
+        if (toolUse.name === "Weather_Tool") {
+          try {
+            let jsonData;
+            const current_weather = await callWeatherTool(
+              longitude,
+              latitude,
+            ).then((current_weather) => current_weather);
+            const currentWeather = current_weather;
+            const toolResult = {
+              toolResult: {
+                toolUseId: toolUseID,
+                content: [{ json: currentWeather }],
+              },
+            };
+            toolResultFinal.push(toolResult);
+          } catch (err) {
+            const toolResult = {
+              toolUseId: toolUseID,
+              content: [{ json: { text: err.message } }],
+              status: "error",
+            };
+          }
+        }
+      }
+    }
+    //
+    const toolResultMessage = {
+      role: "user",
+      content: toolResultFinal,
+    };
+    messages.push(toolResultMessage);
+    // Send the conversation to Amazon Bedrock
+    await ProcessModelResponseAsync(
+      await SendConversationtoBedrock(messages),
+      messages,
+    );
+  } catch (error) {
+    const errorString = error.toString();
+    const searchString =
+      "ValidationException: The toolResult blocks at messages.2.content contain duplicate Ids:";
```

Comment:
The issue with the loop for tool use is what I think is causing this error. Once that's fixed we should be able to remove this part

Reply:
Done